File: src/main/java/org/umundo/model/Question.java
package org.umundo.model;

import org.umundo.core.Message;

public class Question {

  private int questionId;
  private String question;
  private String answerA;
  private String answerB;
  private String answerC;
  private String answerD;
  private int correctAnswer;

  public Question(int questionId, String question, String answerA, String answerB, String answerC, String answerD, int correctAnswer) {
    this.questionId = questionId;
    this.question = question;
    this.answerA = answerA;
    this.answerB = answerB;
    this.answerC = answerC;
    this.answerD = answerD;
    this.correctAnswer = correctAnswer;
  }

  public static Question fromMessage(Message m) {
    return new Question(
        Integer.parseInt(m.getMeta("id")),
        m.getMeta("question"),
        m.getMeta("answerA"),
        m.getMeta("answerB"),
        m.getMeta("answerC"),
        m.getMeta("answerD"),
        Integer.parseInt(m.getMeta("correctAnswer"))
    );
  };

  public Message get() {
    Message m = new Message();
    m.putMeta("type", "question");
    m.putMeta("id", questionId + "");
    m.putMeta("question", question);
    m.putMeta("answerA", answerA);
    m.putMeta("answerB", answerB);
    m.putMeta("answerC", answerC);
    m.putMeta("answerD", answerD);
    m.putMeta("correctAnswer", correctAnswer + "");
    return m;
  }

  public int getQuestionId() {
    return questionId;
  }

  public String getQuestion() {
    return question;
  }

  public String getAnswerA() {
    return answerA;
  }

  public String getAnswerB() {
    return answerB;
  }

  public String getAnswerC() {
    return answerC;
  }

  public String getAnswerD() {
    return answerD;
  }

  public int getCorrectAnswer() {
    return correctAnswer;
  }
}

File: pyQuiz/question.py
import config

class Question:
    """Represents a multiple choice question."""

    def __init__(self, question, question_id):
        self._q = question
        self._qid = question_id

    def getQuestion(self):
        """String representation of the question"""
        return self._q["question"]

    def getQuestionId(self):
        """An identifier, unique to the localy loaded questions"""
        return self._qid

    def getAnswerA(self):
        """String representation of the first possible answer"""
        return self._q["A"]

    def getAnswerB(self):
        """String representation of the second possible answer"""
        return self._q["B"]

    def getAnswerC(self):
        """String representation of the third possible answer"""
        return self._q["C"]

    def getAnswerD(self):
        """String representation of the forth possible answer"""
        return self._q["D"]

    def getCorrectAnswer(self):
        """A number between 0-3 which identifies the correct answer to this question"""
        return self._q["correctAnswer"]

    def getTimeout(self):
        """The available time to answer this question in milliseconds"""
        return config.QUESTION_TIME_MS

    def toDict(self):
        """
        Compatible representation of this object as dictonary.
        This representation should be used whenever the object is send over the
        wire via a umundo message.
        """
        return {
            "type": config.Message.QUESTION,
            "id": self._qid,
            "question": self._q["question"],
            "answerA": self._q["A"],
            "answerB": self._q["B"],
            "answerC": self._q["C"],
            "answerD": self._q["D"],
            "correctAnswer": self._q["correctAnswer"],
            "timeout": config.QUESTION_TIME_MS,
        }

    def fromMsg(msg):
        """
        Generates a question object from a umundo message.
        """
        return Question({
            "question": msg.getMeta("question"),
            "A": msg.getMeta("answerA"),
            "B": msg.getMeta("answerB"),
            "C": msg.getMeta("answerC"),
            "D": msg.getMeta("answerD"),
            "correctAnswer": msg.getMeta("correctAnswer"),
        }, msg.getMeta("id"))

